src/udev/udev-builtin-net_id.c

```diff
@@ -161,6 +167,63 @@ static struct udev_device *skip_virtio(struct udev_device *dev) {
         return parent;
 }
 
+static int get_virtfn_info(struct udev_device *dev, struct netnames *names, struct virtfn_info *vf_info) {
+        struct udev *udev;
+        const char *physfn_link_file;
+        _cleanup_free_ char *physfn_pci_syspath = NULL;
+        _cleanup_free_ char *virtfn_pci_syspath = NULL;
+        struct dirent *dent;
+        _cleanup_closedir_ DIR *dir = NULL;
+        struct virtfn_info vf_info_local;
+        int rc = -ENOENT;
+
+        udev = udev_device_get_udev(names->pcidev);
+        if (!udev)
+                goto out;
+        /* Check if this is a virtual function. */
+        physfn_link_file = strjoina(udev_device_get_syspath(names->pcidev), "/physfn");
+        if (chase_symlinks(physfn_link_file, NULL, 0, &physfn_pci_syspath) <= 0)
+                goto out;
+
+        /* Get physical function's pci device. */
+        vf_info_local.physfn_pcidev = udev_device_new_from_syspath(udev, physfn_pci_syspath);
+        if (!vf_info_local.physfn_pcidev)
+                goto out;
+
+        /* Find the virtual function number by finding the right virtfn link. */
+        vf_info_local.suffix[0] = '\0';
+        dir = opendir(physfn_pci_syspath);
+        if (!dir)
+                goto out_unref;
+        FOREACH_DIRENT_ALL(dent, dir, break) {
+                _cleanup_free_ char *virtfn_link_file = NULL;
+                if (!startswith(dent->d_name, "virtfn"))
+                        continue;
+                virtfn_link_file = strjoin(physfn_pci_syspath, "/", dent->d_name);
+                if (!virtfn_link_file) {
+                        rc = -ENOMEM;
+                        goto out_unref;
+                }
+                if (chase_symlinks(virtfn_link_file, NULL, 0, &virtfn_pci_syspath) < 0)
+                        continue;
+                if (streq(udev_device_get_syspath(names->pcidev), virtfn_pci_syspath)) {
+                        if (!snprintf_ok(vf_info_local.suffix, sizeof(vf_info_local.suffix), "v%s", &dent->d_name[6]))
+                                goto out_unref;
+                        break;
+                }
+        }
+        if (isempty(vf_info_local.suffix))
```

Comment:
vf_info_local is possibly never initialized, hence you check for uninitialized memory here